tests/ampi/migration/test.C

```diff
@@ -1,63 +1,92 @@
 /**
- * AMPI Thread Migration Test
- * Migrate the rank 1 AMPI process from node to node in order.
+ * AMPI Migration Test:
+ * Migrate AMPI rank 1 from PE to PE in order.
  */
 #include <stdio.h>
 #include <stdlib.h>
+#include <vector>
+#include <cassert>
 #include "mpi.h"
-#include "charm.h" /* For CkAbort */
 
+int get_my_pe() {
+  int flag;
+  int *my_wth;
+  MPI_Comm_get_attr(MPI_COMM_WORLD, AMPI_MY_WTH, &my_wth, &flag);
+  assert(flag == 1);
+  return *my_wth;
+}
+
+int get_num_pes() {
+  int flag;
+  int *num_wths;
+  MPI_Comm_get_attr(MPI_COMM_WORLD, AMPI_NUM_WTHS, &num_wths, &flag);
+  assert(flag == 1);
+  return *num_wths;
+}
 
-int main(int argc,char **argv)
+int main(int argc, char **argv)
 {
+  int rank;        /* my rank # */
+  int p;           /* number of ranks */
+  int my_init_pe;  /* my initial PE according to mapfile */
 
-  int rank;            /* process id */
-  int p;                /* number of processes */
+  MPI_Init(&argc, &argv);
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+  MPI_Comm_size(MPI_COMM_WORLD, &p);
+  int my_pe = get_my_pe();
 
-  MPI_Init( &argc, &argv );
-  MPI_Comm_rank( MPI_COMM_WORLD, &rank);
-  MPI_Comm_size( MPI_COMM_WORLD, &p );
+  // Test that mapfile is being used correctly:
+  if (argc == 2) {
```

Comment:
Shouldn't this be `(argc >= 2 && !strcmp(argv[1], "--test_mapfile"))`?

Reply:
Fixed.